File: result/metagpt/claude/humaneval/43.c
#include <stdbool.h>

bool pairs_sum_to_zero(int* l, int size) {
    for (int i = 0; i < size; i++) {
        for (int j = i + 1; j < size; j++) {
            if (l[i] + l[j] == 0) {
                return true;
            }
        }
    }
    return false;
}

File: result/metagpt/deepseek/humaneval/43.rs
pub fn pairs_sum_to_zero(numbers: &[i32]) -> bool {
    for i in 0..numbers.len() {
        for j in i+1..numbers.len() {
            if numbers[i] + numbers[j] == 0 {
                return true;
            }
        }
    }
    false
}

#[cfg(test)]
mod tests {
    use super::*;

    #[test]
    fn test_pairs_sum_to_zero() {
        assert!(pairs_sum_to_zero(&[1, -1, 2]));
        assert!(!pairs_sum_to_zero(&[1, 2, 3]));
    }
}